src/index.ts

```diff
@@ -321,32 +323,58 @@ interface ConvertOptions {
 export async function convert(
   input: string,
   outputPaths?: string | string[],
-  { to, from, encodeOptions }: ConvertOptions = {}
+  { to, from, encodeOptions = defaultEncodeOptions }: ConvertOptions = {}
 ): Promise<string | undefined> {
   let outputPaths_: (string | undefined)[]
   if (outputPaths === undefined) outputPaths_ = [undefined]
   else if (typeof outputPaths === 'string') outputPaths_ = [outputPaths]
   else outputPaths_ = outputPaths
 
+  const { shouldZip } = encodeOptions
+
   const inputFile = vfile.create(input)
   const node = await decode(inputFile, input, from)
 
   let index = 0
+  const files: string[] = []
   for (const outputPath of outputPaths_) {
     const outputFile = await encode(node, {
       ...defaultEncodeOptions,
       format: to,
       filePath: outputPath,
       ...encodeOptions
     })
-    if (outputPath !== undefined) await vfile.write(outputFile, outputPath)
+
+    if (outputPath !== undefined) {
+      // Write file to path (note that this may have been done already by
+      // the encoding codec, in which case `outputFile.contents === undefined`
+      // and this is effectively a no-op)
+      await vfile.write(outputFile, outputPath)
+
+      // Record files generated
+      files.push(outputPath)
+      // The convention amongst codecs is to put media files in
+      // a sibling folder with `.media` appended to the name
+      // We rely on that convention here...
+      const mediaFolder = outputPath + '.media'
```

Comment:
I'd like to see '`.media'` moved to a variable at some stage so its consistent throughout.